sdk/core/azure-core/tests/test_exceptions.py

```diff
@@ -104,3 +123,38 @@ def test_httpresponse_error_with_response(self):
         assert error.reason == 'OK'
         assert error.status_code == 200
         assert error.error is None
+
+    def test_odata_v4_exception(self):
+        message = {
+            "error": {
+                "code": "501",
+                "message": "Unsupported functionality",
+                "target": "query",
+                "details": [{
+                    "code": "301",
+                    "target": "$search",
+                    "message": "$search query option not supported",
+                }],
+                "innererror": {
+                    "trace": [],
+                    "context": {}
+                }
+            }
+        }
+        exp = ODataV4Error(_build_response(json.dumps(message).encode("utf-8")))
+
+        assert exp.code == "501"
+        assert exp.message == "Unsupported functionality"
+        assert exp.target == "query"
+        assert exp.details[0].code == "301"
+        assert exp.details[0].target == "$search"
+        assert "trace" in exp.innererror
+        assert "context" in exp.innererror
+
+        message = {}
+        exp = ODataV4Error(_build_response(json.dumps(message).encode("utf-8")))
+        assert exp.message == "Operation returned an invalid status 'Bad Request'"
+
+        exp = ODataV4Error(_build_response(b""))
+        assert exp.message == "Operation returned an invalid status 'Bad Request'"
+        assert str(exp) == "Operation returned an invalid status 'Bad Request'"
```

Comment:
Maybe tests to cover `json_object["details"]` being empty (or missing) and `json_object["target"]` missing?

Reply:
test pushed, thanks!